File: us_jaggery/public/js/account_tree.js
// Show the custom Auto-ID next to each account in the Chart of Accounts tree.
// Patches frappe.treeview_settings["Account"] (defined by erpnext/.../account_tree.js):
//  - get_tree_nodes -> our provider that also returns custom_auto_id
//  - get_label      -> renders "<account name> [Auto-ID]" for ledger accounts
frappe.provide("frappe.treeview_settings");

window.us_jaggery_patch_account_tree = function () {
	const s = frappe.treeview_settings && frappe.treeview_settings["Account"];
	if (!s || s.__us_jaggery_patched) return;

	s.get_tree_nodes = "us_jaggery.overrides.account_tree.get_children";

	s.get_label = function (node) {
		const data = (node && node.data) || {};
		const label = frappe.utils.escape_html(data.value || node.label || "");
		const id = data.custom_auto_id;
		if (id) {
			return (
				label +
				' <span class="text-muted" style="font-weight:600">[' +
				frappe.utils.escape_html(id) +
				"]</span>"
			);
		}
		return label;
	};

	s.__us_jaggery_patched = true;
};

us_jaggery_patch_account_tree();
$(document).on("page-change", us_jaggery_patch_account_tree);

File: us_jaggery/overrides/account_tree.py
"""Chart of Accounts tree node provider that also returns the custom Auto-ID.

Wraps ERPNext's get_children so each node carries ``custom_auto_id``; the tree's
client-side get_label (public/js/account_tree.js) renders it next to the account.
Wired by setting frappe.treeview_settings["Account"].get_tree_nodes to this path.
"""

import frappe


@frappe.whitelist()
def get_children(doctype, parent=None, company=None, is_root=False):
	from erpnext.accounts.utils import get_children as erpnext_get_children

	nodes = erpnext_get_children(doctype, parent, company, is_root)
	if doctype == "Account" and nodes:
		names = [n.get("value") for n in nodes]
		id_map = dict(
			frappe.get_all(
				"Account", filters={"name": ["in", names]},
				fields=["name", "custom_auto_id"], as_list=True
			)
		)
		for n in nodes:
			n["custom_auto_id"] = id_map.get(n.get("value")) or ""
	return nodes
